src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MemberAnalysisResult.cs

```diff
@@ -14,6 +17,7 @@ internal struct MemberAnalysisResult
         public readonly ImmutableArray<Conversion> ConversionsOpt;
         public readonly ImmutableArray<int> BadArgumentsOpt;
         public readonly ImmutableArray<int> ArgsToParamsOpt;
+        public readonly ArrayBuilder<TypeParameterDiagnosticInfo> ConstraintFailureDiagnosticsOpt;
```

Comment:
I'd be tempted to make `ConstraintFailureDiagnosticsOpt` immutable like the other members.